python/ray/tune/tune.py

```diff
@@ -709,22 +699,68 @@ class and registered trainables.
         local_path=local_path, remote_path=remote_path, sync_config=sync_config
     )
 
-    checkpoint_score_attr = checkpoint_score_attr or ""
-    if checkpoint_score_attr.startswith("min-"):
-        checkpoint_score_attr = checkpoint_score_attr[4:]
-        checkpoint_score_order = "min"
-    else:
-        checkpoint_score_order = "max"
-
-    checkpoint_config = CheckpointConfig(
-        num_to_keep=keep_checkpoints_num,
-        checkpoint_score_attribute=checkpoint_score_attr,
-        checkpoint_score_order=checkpoint_score_order,
-        checkpoint_frequency=checkpoint_freq,
-        checkpoint_at_end=checkpoint_at_end,
-        _checkpoint_keep_all_ranks=checkpoint_keep_all_ranks,
-        _checkpoint_upload_from_workers=checkpoint_upload_from_workers,
-    )
+    checkpoint_config = checkpoint_config or CheckpointConfig()
+
+    # For backward compatibility
+    # TODO(jungong): remove after 2.7 release.
+    if keep_checkpoints_num is not None:
+        warnings.warn(
+            "keep_checkpoints_num is deprecated and will be removed. "
+            "use checkpoint_config.num_to_keep instead.",
+            DeprecationWarning,
+        )
+        checkpoint_config.num_to_keep = keep_checkpoints_num
+    if checkpoint_score_attr is not None:
+        warnings.warn(
+            "checkpoint_score_attr is deprecated and will be removed. "
+            "use checkpoint_config.checkpoint_score_attribute instead.",
+            DeprecationWarning,
+        )
+
+        if checkpoint_score_attr.startswith("min-"):
+            warnings.warn(
+                "using min- and max- prefixes to specify checkpoint score "
+                "order is deprecated. Use CheckpointConfig.checkpoint_score_order "
+                "instead",
+                DeprecationWarning,
+            )
+            checkpoint_config.checkpoint_score_attribute = checkpoint_score_attr[4:]
+            checkpoint_config.checkpoint_score_order = "min"
+        else:
+            checkpoint_config.checkpoint_score_attribute = checkpoint_score_attr
+            checkpoint_config.checkpoint_score_order = "max"
+
+        checkpoint_config.score_attr = checkpoint_score_attr
+    if checkpoint_freq > 0:
+        warnings.warn(
+            "checkpoint_freq is deprecated and will be removed. "
+            "use checkpoint_config.checkpoint_frequency instead.",
+            DeprecationWarning,
+        )
+        checkpoint_config.checkpoint_frequency = checkpoint_freq
+    if checkpoint_at_end:
+        warnings.warn(
+            "checkpoint_at_end is deprecated and will be removed. "
+            "use checkpoint_config.checkpoint_at_end instead.",
+            DeprecationWarning,
+        )
+        checkpoint_config.checkpoint_at_end = checkpoint_at_end
```

Comment:
The current default is None, so if the user passes in `False`, that would get ignored, and the `checkpoint_at_end` would stay None and default to True for class trainables.

```suggestion
    if checkpoint_at_end is not None:
        warnings.warn(
            "checkpoint_at_end is deprecated and will be removed. "
            "use checkpoint_config.checkpoint_at_end instead.",
            DeprecationWarning,
        )
        checkpoint_config.checkpoint_at_end = checkpoint_at_end
```

Reply:
Ah, it's actually `False` in `tune.run()` - I think this check is fine. I also don't think many users explicitly pass `False` here